src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -186,16 +184,15 @@ impl Executor for IndexScanExecutor {
         self.count = 0;
     }
 
-    fn collect_statistics_into(&mut self, statistics: &mut Statistics) {
-        statistics.add(&self.statistics);
-        self.statistics = Statistics::default();
+    fn collect_metrics_into(&mut self, metrics: &mut CopMetrics) {
```

Comment:
How about make `collect_metrics_into` move `self` to avoid `first_collect`?

Reply:
Would we collect_metrics more than once in streaming? If not, we can move self here.